src/clincoded/static/components/collection.js

```diff
@@ -36,370 +49,547 @@ var Collection = module.exports.Collection = createReactClass({
-            loading_or_total = <i className="icon icon-refresh icon-spin"></i>;
-        } else {
-            loading_or_total = <span>Displaying {matching.length} of {total} records</span>;
+            this.setState({ sortColumn, reversed });
+            return rowsCopy;
         }
+    },
 
-        return (
-            <div>
-                <div className="table-meta">
-                    <div className="container">
-                        <div className="row table-summary">
-                            <div className="col-sm-6 table-count">
-                                {loading_or_total}
-                            </div>
-                            <form ref="form" className="form-inline col-sm-6 table-filter" onKeyUp={this.handleKeyUp} 
-                                data-skiprequest="true" data-removeempty="true">
-                                <div className="form-group table-filter-input">
-                                    <label htmlFor="table-filter">Filter table by:</label>
-                                    <input ref="q" disabled={this.state.communicating || undefined} 
-                                        name="q" type="search" defaultValue={searchTerm} 
-                                        className="form-control" id="table-filter" /> 
-                                    <i className="icon icon-times-circle clear-input-icon" hidden={!searchTerm} onClick={this.clearFilter}></i>
-                                </div>
-                                <input ref="sorton" type="hidden" name="sorton" defaultValue={sortOn !== defaultSortOn ? sortOn : ''} />
-                                <input ref="reversed" type="hidden" name="reversed" defaultValue={!!reversed || ''} />
-                            </form>
-                        </div>
-                    </div>
-                </div>
-                <div className="container">
-                    <div className="table-responsive">
-                        <table className="table table-striped table-bordered table-condensed sticky-area">
-                            <thead className="sticky-header">
-                                <tr className="col-headers">
-                                    {headers}
-                                </tr>
-                            </thead>
-                            <tbody>
-                                {rows}
-                            </tbody>
-                        </table>
-                    </div>
-                </div>
-            </div>
-        );
+    clearFilter: function (event) {
+        this.refs.q.value = '';
+        this.submitTimer = setTimeout(this.submit);
     },
 
-    componentDidMount: function () {
-        this.setState({
-            data: this.extractData(this.props),
-            communicating: this.fetchAll(this.props),
-            mounted: true
-        });
+    handleKeyUp: function (event) {
+        if (typeof this.submitTimer != 'undefined') {
```

Comment:
Should this be `!== undefined` similar to line `284` below?

Reply:
Yeah. Some of these functions only show up as changes because I moved them to another part of the file. And I didn't modify the functions themselves. Should I just go through the file and change all of these?